### ISSUE TITLE
Backend assertion failure (possibly from wrong mouse event handling)

### ISSUE THREAD

**Issue by ruby3141** [veeenu/hudhook#181]
![Screenshot 2024-03-31 232953 a](https://github.com/veeenu/hudhook/assets/3256382/5b164cff-8a67-4661-8062-a8d383db2851)

from imgui.cpp:1390 (on "docking" branch. For mainline it's on :1354),

```cpp
IM_ASSERT(!ImGui::IsAliasKey(key)); // Backend cannot submit ImGuiKey_MouseXXX values they are automatically inferred from AddMouseXXX() events.
```

----

Target game is [DJMAX RESPECT V](https://store.steampowered.com/app/960170/DJMAX_RESPECT_V/), which doesn't use any mouse input. \
I couldn't find the exact condition to reproduce the crash. still testing.

**Comment by vSylva** [comment]
https://github.com/veeenu/hudhook/commit/f058304463a28e0bd328dc35c85fabaeb0559bee

Try this

**Comment by veeenu** [comment]
What `hudhook` version are you on? And how are you using the docking branch? This issue occurred before but it has now been fixed and a new release with the fix is on crates.io.

**Comment by ruby3141** [comment]
Oh It was v0.6.1...
I started my project about a month ago with that version,
and thought I'm still using newest release because of github's stupid "3 weeks ago".

Sorry for wasting your time.

**Comment by veeenu** [comment]
All inputs on our end are correctly passed through `add_mouse_event`. I do not know where that assertion could be triggered. Are you by chance doing `ui.io_mut()[something] = ...` in your code? That is discouraged and will trigger the assertion above.

**Comment by ruby3141** [comment]
As far as I know `imgui::Ui` doesn't have `io_mut()` method so it's certainly not happening.

For `imgui::Context::io_mut()` which can be accessed on `before_render()`,
there's some logic to update `Io.delta_time` and `Io.mouse_draw_cursor`.

**Comment by veeenu** [comment]
Right about `ui.io_mut`, it's not a thing.

Have you tried what happens if you don't perform that logic? Please try to identify if any changes on your end result in this behavior going away so we can pinpoint the problem.

**Comment by ruby3141** [comment]
Removed everything except hooking part but crash still happens.
```rs
use hudhook::{hooks::dx11::ImguiDx11Hooks, ImguiRenderLoop};

struct BlankContext;

impl ImguiRenderLoop for BlankContext {
    fn render(&mut self, _ui: &mut imgui::Ui) {}
}

hudhook::hudhook!(ImguiDx11Hooks, BlankContext);
```

**Comment by veeenu** [comment]
You mentioned the docking branch in your first message. How are you accessing that? Are you compiling against a different underlying version of `imgui`?

I reviewed our code but I couldn't find code paths that can ever hit that assertion, and also I do not have access to that game to test it myself. I need more information, a lot of it, if we hope to find out anything about this at all.

I encourage you to clone the repo locally and introduce log statements nearby the input related code, or at least implement the `on_window_proc` method and log what messages are received by the game window, and report your findings. Thank you!

**Comment by ruby3141** [comment]
After the first close of this issue I opt out for imgui docking because it seems unnecessary to me.
I accessed it via `docking` feature of `imgui` crate by the way.
I didn't test anything about the feature, so I've no idea it would work well with hudhook or not.

I will add log and collect info about why did it happen.
Slow progression expected since I'm not familiar with rust. (was C# main)

**Comment by veeenu** [comment]
There's no rush! Take your time. This is open source, it's all based on volunteering whatever time one wants or can. :grin:

**Comment by ruby3141** [comment]
I thought current release v0.6.5 includes commit f058304 (fix for #172).
Found out it doesn't, while checking for #176 breaking change on latest source.
<sub>I recompiled and checked over ten times with denial.</sub>

With `hudhook = { git = "https://github.com/veeenu/hudhook.git", rev = "f058304463a28e0bd328dc35c85fabaeb0559bee" }`, problem is gone.
So ultimately I was trying to re-solve already fixed problem for two weeks. 🤦‍♂️

Thank you for your help.

**Comment by veeenu** [comment]
Huh! I thought it did, actually. My bad. Planning to release `0.7` at some point soonish so that fix can get in. Thank you for reporting this.
